### ISSUE TITLE
fitBetaColoc error 

### ISSUE THREAD

**Issue by jianvhuang** [mikelove/mrlocus#4]
fitBetaColoc returned an error as described below. Do you have any advice?

```
I used the below command and successfully obtained data2.

data2 <- flipAllelesAndGather(data1$sum_stat, data1$ld_mat,
                              a="eqtl", b="gwas",
                              ref="ref", eff="effect",
                              beta="beta", se="se",
                              snp_id="snpid", sep="_",
                              a2_plink="effect_eqtl",
                              ab_last=TRUE,
                              plot=FALSE)

> names(data2)
[1] "beta_hat_a" "beta_hat_b" "se_a"       "se_b"       "Sigma"
[6] "alleles"

```
```

However, when I moved on to the coloc stage, an error was returned.
Also, unlike the simulated example, I only have data2$Sigma, not data2$Sigma_a and data2$Sigma_b. 

> coloc_fit <- list()
> nclust <- length(data2$beta_hat_a)
> lengths(data2$beta_hat_a)
[1] 26  9
>
>
>
> options(mc.cores=2)
> for (j in 1:nclust) {
+   coloc_fit[[j]] <- with(data2,
+                          fitBetaColoc(beta_hat_a = beta_hat_a[[j]],
+                                       beta_hat_b = beta_hat_b[[j]],
+                                       se_a = se_a[[j]],
+                                       se_b = se_b[[j]],
+                                       Sigma_a = Sigma[[j]],
+                                       Sigma_b = Sigma[[j]]))
+ }
Error: $ operator is invalid for atomic vectors

```

```
I checked the source code on GitHub, and found that the error was returned at the below line in the fitBetaColoc function. 
I suspect it is due to the rstan package. But I have the same version as showed in the vignette. 

stanfit <- rstan::sampling(stanmodels$beta_coloc, data, ...)
```

```

> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /rds/general/user/jhuang18/home/anaconda3/envs/jhuang/lib/R/lib/libRblas.so

locale:
[1] en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] mrlocus_0.0.22       devtools_2.3.2       usethis_2.0.1
 [4] ieugwasr_0.1.5       TwoSampleMR_0.5.6    stringr_1.4.0
 [7] data.table_1.12.2    rstan_2.21.2         ggplot2_3.3.2
[10] StanHeaders_2.21.0-7 V8_3.4.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6                   lattice_0.20-38
 [3] prettyunits_1.0.2            ps_1.6.0
 [5] rprojroot_1.3-2              assertthat_0.2.1
 [7] glmnet_4.0-2                 digest_0.6.18
 [9] foreach_1.5.0                R6_2.4.0
[11] backports_1.1.4              mr.raps_0.2
[13] stats4_3.6.1                 pillar_1.4.6
[15] rlang_0.4.10                 curl_3.3
[17] rstudioapi_0.13              callr_3.6.0
[19] Matrix_1.2-17                desc_1.3.0
[21] genetics.binaRies_0.0.0.9000 labeling_0.3
[23] splines_3.6.1                loo_2.4.1
[25] munsell_0.5.0                compiler_3.6.1
[27] pkgconfig_2.0.2              pkgbuild_1.2.0
[29] shape_1.4.4                  tcltk_3.6.1
[31] tidyselect_1.1.0             tibble_3.0.3
[33] gridExtra_2.3                codetools_0.2-16
[35] matrixStats_0.57.0           crayon_1.3.4
[37] dplyr_1.0.2                  withr_2.4.1
[39] MASS_7.3-53                  grid_3.6.1
[41] jsonlite_1.7.2               gtable_0.3.0
[43] lifecycle_1.0.0              magrittr_1.5
[45] scales_1.0.0                 RcppParallel_5.1.4
[47] cli_2.4.0                    stringi_1.4.3
[49] cachem_1.0.3                 remotes_2.3.0
[51] fs_1.3.1                     testthat_3.0.2
[53] ellipsis_0.3.1               generics_0.0.2
[55] vctrs_0.3.2                  nortest_1.0-4
[57] iterators_1.0.10             tools_3.6.1
[59] glue_1.4.1                   purrr_0.3.4
[61] pkgload_1.2.1                fastmap_1.1.0
[63] processx_3.5.1               parallel_3.6.1
[65] survival_2.44-1.1            inline_0.3.15
[67] colorspace_1.4-1             sessioninfo_1.1.1
[69] memoise_2.0.0
>

```

**Comment by mikelove** [comment]
Thanks for the report. Can you run `traceback()` after the error?

**Comment by mikelove** [comment]
Another question: how did you install `mrlocus`? If you are correct about `stanmodels` it would be that the model was not built / is not available in your session.

**Comment by jianvhuang** [comment]
Thank you. I think the issue is solved.

I previously download the github package and source the R script. 
The coloc step can be performed after I reinstalled the package using the below script. 
```
library(devtools)
install_github("mikelove/mrlocus")

```

**Comment by mikelove** [comment]
Got it, yeah installation needed so that the stan models are compiled from C++ and available to the R functions.

Let me know if you get stuck again.

Also, as one of the first MRLocus users (as far as I know) I see you have *TwoSampleMR* loaded. Would you prefer if MRLocus were able to take TwoSampleMR data objects as input? E.g. the output from:

https://mrcieu.github.io/TwoSampleMR/reference/harmonise_data.html

**Comment by jianvhuang** [comment]
I currently follow the below workflow.
1) extract the cis-eqtl for a target gene 
2) find the clumps for the selected SNPs based on 1000G Eur reference (using PLINK)
3) estimate the LD matrix for each clump using TwoSampleMR package in R 
4) perform _collapseHighCorSNPs_ and _flipAllelesAndGather_

If MRLocus were able to take the output from the _harmonise_data_ function, I think the workflow might change to 
1) extract the cis-eqtl for a target gene 
2) use _harmonise_data_ function in R to harmonise eqtl and gwas datasets
3) find the clumps for the harmonise SNPs (I don't think TwoSampleMR package can return clumps, so this step will still need to be performed using PLINK)
4) estimate the LD matrix for each clump using TwoSampleMR package in R 


Accepting the output from the _harmonise_data_ function may help the TwoSampleMR users to incorporate MRLocus into their current MR pipeline. But for me, who is more familiar with R, these two workflows require a similar effort.  What will be more helpful may be a function to identify clumps in R, so the user will not need to switch between R and PLINK.

**Comment by mikelove** [comment]
Got it, makes perfect sense. So I'll wait until users ask for `harmonise_data` integration, because as it stands, we've duplicated their harmonization functionality (I didn't realize this until I had already put together the software).

Re: identifying clumps, I'm not aware yet of a way to do this from R. It seems also like usually people have this data in a `bfile` so unless there is a function to call PLINK from R, I think your pipeline is the best way (and what we used).

I predict that going forward, the independent signals will be identified by the QTL study analysts using conditioning, and output by software such as APEX. Then variance covariance files will provide sliding window information on covariance of betas that downstream software like MRLocus could take as input. I'd be happy to add this type of functionality once there is a user asking for it.
